158619400 edit ride

xcixor opened the pull request:
#### Purpose of the PR
* Create endpoint to edit a ride
#### Tasks to be completed
* Adds functionality to edit a ride
* Creates an endpoint to edit a ride
#### Manual testing
* Navigate to the repo's main page via  [this link](https://github.com/xcixor/my-ride)
* Under the repo's name copy the clone's url
* Open a terminal and navigate to where you want the repo to be downloaded
* Type git clone 'the url you copied above' and press enter to download
* Set up a virtual env and install the dependencies in requirements.txt
* Checkout to the branch 158619400-edit-ride
* run the run.py file to start the server
* Make a request using postman with fields in your ride that you want to edit
#### Running the test
* run the test with pytest like so -> py.test tests/
##### Tests running 
* All rides test
* All controller tests
* All user tests 
##### Tests failing
* Endpoint tests
#### Relevant pivotal tracker stories 
[#158619400](https://www.pivotaltracker.com/n/projects/2179559)
#### Screenshots
##### ride before edit
![ridebefore](https://user-images.githubusercontent.com/22756288/41914242-1eb4a21c-795b-11e8-911b-3b16c8d59872.png)
#### ride after
![rideafter](https://user-images.githubusercontent.com/22756288/41914265-2ddce1b4-795b-11e8-9a3b-e84808d3e8c2.png)

xcixor committed:
Ignore editor settings

xcixor committed:
feat(register user)

-Implement the model for the user

-Manipulate the model for registration in the controller

-Add endpoint for user registratio in the views

-Register the route in init file

xcixor committed:
-Edit travis to fix build issues

xcixor committed:
feat(login user)

-Add functionality to login user

-Register login route

xcixor committed:
maintain(commented code)

-Remove unused code

[Finishes #158580942]

xcixor committed:
feat(create ride: #158549088)

-Implement function for creating a ride

-Manipulate logic for creating a ride in the controller

-Add endpoint for adding a ride in the views

xcixor committed:
maintain(tests)

-Remove errors in tests

xcixor committed:
feat(get all rides)

-Add function to retrieve all events

-Add endpoint to retrieve all events

xcixor committed:
maintain(tests)

-Correct typing errors in tests

[Finishes #158549001]

xcixor committed:
feat(fetch ride)

-Add functionality to get a ride

-Create endpoint to fetch a ride

xcixor committed:
feat(make reques)

-Add functionality to make request in the model and controller

-Add endpoint to make reques

xcixor committed:
feate(edit ride)

-Add functionality to edit a ride

meshack-mbuvi commented on app/api_1_0/controller.py:
```diff
@@ -43,29 +43,25 @@ def generate_id(item_data, item_id=0):
+    #         else:
+    #             return {'Status': False, 'Message': 'Password incorrect!'}
+    #     else:
+    #         return {'Status': True, 'Message': 'User does not exist'}
```
remove code you have commented out

meshack-mbuvi commented on app/api_1_0/models.py:
```diff
@@ -57,3 +57,25 @@ def verify_password_length(password):
+    def login(self, email, password):
+        """Check if credentials are correct to allow login."""
+        res = self.get_user(email)
+        print('&&&&&&&&&&&&&&&&&&&&&&', res.get('Message'))
```
remember to remove print statement;It won't be accessible to any user

meshack-mbuvi commented on .gitignore:
```diff
@@ -7,3 +7,4 @@ wireframes
 *pytest_cache/*
 *__pycache__*
 *pytest_cache*
```
Add .gitignore to .gitignore file., should not be pushed

artorious replied:
Good to know @meshack-mbuvi  will take that advice. Thank you.
@xcixor your progress is inspiring

artorious replied:
Hi @meshack-mbuvi? Question.....if `.gitignore` is not pushed, doesn't that mean that all those who `clone` and `fork` the project need to add one if they're to keep their version clean? Advise.

LWanjiru replied:
@meshack-mbuvi I am curious as to what informs this opinion "Add .gitignore to .gitignore file., should not be pushed" ... Please look into the following resource on this **cc** @artorious  https://help.github.com/articles/ignoring-files/

meshack-mbuvi replied:
@LWanjiru Seen and read

meshack-mbuvi reviewed (commented):
Looks fine

LWanjiru commented:
Why is the Travis build failing? @xcixor

artorious commented:
I like what I see. Your progress is inspiring

LWanjiru commented on app/api_1_0/controller.py:
```diff
@@ -0,0 +1,114 @@
+                return {'Status': False, 'Message': result.get('Message')}
+
+    @staticmethod
+    def generate_id(item_data, item_id=0):
```
This function is not explicit, which `list of items` does it cater to? (considering our project is about drivers, passengers and rides)

xcixor replied:
I agree it has no relation with the class let fix that

xcixor committed:
feat(api endpoints)

-Add functionality to logout

-Add functionality to edit ride

-refactorr test to ensure run

xcixor commented:
@LWanjiru , had not fixed the bug I was facing yesterday, fixed it and travis is all good

xcixor committed:
-Add cov for py.test in requirements.txt to fix build

xcixor committed:
-maintenance(code quality)

-Reformat code to enhance readability

-update badges to reflect current working branch

-Update readme.md to offer more details about app

xcixor committed:
-fix(whitespace)

-Remove whitespace from method return

xcixor committed:
-Add procfile for heroku deployment

xcixor committed:
-Update readme.md with hosting link

xcixor committed:
-maintain(quality)

-Implement codacy suggestions

xcixor committed:
-fix(travis)

-remove packages resources dependency that breaks build.

xcixor committed:
-Remove unused code to increase quality

xcixor committed:
deploy(heroku)

-edit procfile to fix heroku deploy issues

xcixor committed:
edit procfile

xcixor committed:
fix heroku

LWanjiru commented:
Did you resolve the issues I raised on this PR?

xcixor committed:
-deploy(heroku)

-add port to run app on heroku

xcixor committed:
Add api's documentaion link to README.md

xcixor committed:
rafctor(code quality)

-Edit code to ensure quality

xcixor commented:
I have fixed all the issues @LWanjiru 
As for codacy it seems to have an issue with previous code that I have already fixed. in the branch the grade is fine

xcixor committed:
-Edit badges to reflect current project progress

xcixor committed:
-Update travis to report cverage

xcixor committed:
-revert travis script to fix build

xcixor committed:
-use coveralls to report coverage

xcixor committed:
-report coverage

xcixor committed:
edit codacy badge to reflect current working branch.

xcixor committed:
edit codacy badge to reflect current working branch.

LWanjiru reviewed (changes requested):
Some of your checks are failing @xcixor  please look into this

xcixor committed:
Fix(non explicit method)

-Separate non explicit functionality from class implementation.

xcixor committed:
fix(typo)

-fix typo in readme.md

xcixor committed:
-update git-ignore with unwanted files.

LWanjiru commented on #1#travis.log:
```diff
@@ -0,0 +1,87 @@
+Command line:
```
What does this file do @xcixor ?

xcixor replied:
that travis.log is redundant I had been trying a different method of reporting coverage but abandoned it, let me remove that. As for pylint I did set it up and also included it in my requirements.txt

LWanjiru commented on .codacy.yml:
```diff
@@ -0,0 +1,5 @@
+exclude_paths:
+  - '*.html'
+  - '*.js'
+  - '*.css'
```
Add a new line here

LWanjiru commented on .travis.yml:
```diff
@@ -1,6 +1,8 @@
+- pip install -r requirements.txt
+- pip install coveralls
+script: py.test --cov=app tests/ && coverage report
 after_success: coveralls
```
add a new line

LWanjiru reviewed (changes requested):
Did you set up PYLINT for your project?

xcixor committed:
-remove redundant files

xcixor committed:
- Add new line at end of file

xcixor commented:
I have fixed the failing tests @LWanjiru they are passing now. I set up pylint on the onset and have trying my best to follow PEP8.

LWanjiru commented:
Looks good.
